src/com/facebook/buck/apple/ProvisioningProfileCopyStep.java

```diff
@@ -192,20 +193,23 @@ public StepExecutionResult execute(ExecutionContext context)
     // Copy the actual .mobileprovision.
     filesystem.copy(provisioningProfileSource, provisioningProfileDestination, CopySourceMode.FILE);
 
+    // Create the application identifier from the app ID and bundle ID
+    String appID = bestProfile.get().getAppID().getFirst() + "." + bundleID;
+
     // Merge the entitlements with the profile, and write out.
     if (entitlementsPlist.isPresent()) {
+
       return (new PlistProcessStep(
               filesystem,
               entitlementsPlist.get(),
               Optional.empty(),
               signingEntitlementsTempPath,
               bestProfile.get().getMergeableEntitlements(),
-              ImmutableMap.of(),
+              ImmutableMap.of(APPLICATION_IDENTIFIER, new NSString(appID)),
```

Comment:
@rmaz Why do we now need to override the value for `APPLICATION_IDENTIFIER`? How did this work beforehand? Was it part of the value returned from `getMergeableEntitlements()`?

I think we should preserve the pre-existing behaviour and specify `APPLICATION_IDENTIFIER` as part of the `additionalKeys` parameter to `PlistProcessStep`, rather than as override. My concern here is that it might override a value already set in `entitlementsPlist`. On the other hand, the argument could be made that if those values did differ, the app will fail to launch due to a code signing issue. Still, it might be safer to err on the side of caution and preserve existing behaviour.

Reply:
I'll wind this one back, I am inclined to agree. The current behaviour is to merge with the application-identifier in the entitlements, yes.